core/Test/Tasty/Core.hs

```diff
@@ -311,7 +328,6 @@ after deptype s =
 data TreeFold b = TreeFold
   { foldSingle :: forall t . IsTest t => OptionSet -> TestName -> t -> b
   , foldGroup :: OptionSet -> TestName -> b -> b
```

Comment:
If we must break `TreeFold` anyway, it's probably better to change `foldGroup` to take `ExecutionMode` into account. Would it allow to express `createTestActions` via `foldTestTree`?

Reply:
I don't think so. The current traversal in `Run.hs` not only propagates info up, it also propagates info down (`Seq Dependency`). I don't see an obvious way to achieve that with `foldTestTree` (unless it is radically changed).

Reply:
Hmm, perhaps it doesn't need to. I'll see if I can make a patch and document the consequences.

Reply:
I've played with this a bit more but I can't get it to work. Admittedly, this was a while ago so I don't exactly remember what didn't work out..